docs/diagrams/ModelClassDiagram.puml

```diff
@@ -5,50 +5,52 @@ skinparam arrowColor MODEL_COLOR
 skinparam classBackgroundColor MODEL_COLOR
 
 Package Model as ModelPackage <<Rectangle>>{
-Class "<<interface>>\nReadOnlyAddressBook" as ReadOnlyAddressBook
+Class "<<interface>>\nReadOnlyInternTracker" as ReadOnlyInternTracker
 Class "<<interface>>\nReadOnlyUserPrefs" as ReadOnlyUserPrefs
 Class "<<interface>>\nModel" as Model
-Class AddressBook
+Class InternTracker
 Class ModelManager
 Class UserPrefs
 
-Class UniquePersonList
-Class Person
-Class Address
+Class UniqueApplicationList
+Class InternApplication
+Class Status
 Class Cycle
-Class Name
-Class Phone
-Class Tag
+Class Company
+Class Role
+Class Deadline
+Class Note
 
 Class I #FFFFFF
 }
 
 Class HiddenOutside #FFFFFF
 HiddenOutside ..> Model
 
-AddressBook .up.|> ReadOnlyAddressBook
+InternTracker .up.|> ReadOnlyInternTracker
 
 ModelManager .up.|> Model
 Model .right.> ReadOnlyUserPrefs
-Model .left.> ReadOnlyAddressBook
-ModelManager -left-> "1" AddressBook
+Model .left.> ReadOnlyInternTracker
+ModelManager -left-> "1" InternTracker
 ModelManager -right-> "1" UserPrefs
 UserPrefs .up.|> ReadOnlyUserPrefs
 
-AddressBook *--> "1" UniquePersonList
-UniquePersonList --> "~* all" Person
-Person *--> Name
-Person *--> Phone
-Person *--> Cycle
-Person *--> Address
-Person *--> "*" Tag
+InternTracker *--> "1" UniqueApplicationList
+UniqueApplicationList --> "~* all" InternApplication
+InternApplication *--> Company
```

Comment:
Correct me if I'm wrong.
Does this mean that one Company can be held in multiple InternApplication objects? I'm p sure we made it a thing to ensure there's no duplicate Company in the list